plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenArtifactMatcher.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal;
+
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.Matcher;
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+/**
+ * Matcher that matches for Maven artifacts only, and sets {@link ArtifactCoordinates} in context attributes.
+ *
+ * @since 3.0
+ */
+public class MavenArtifactMatcher
+    extends ComponentSupport
+    implements Matcher
+{
+  @Override
+  public boolean matches(final Context context) {
+    final String path = context.getRequest().getPath();
+    final MavenFacet mavenFacet = context.getRepository().facet(MavenFacet.class);
+    final ArtifactCoordinates coordinates = mavenFacet.getArtifactCoordinatesParser().parsePath(path);
```

Comment:
While its technically possible for matchers to use facets I never imagined the matchers were to be complex enough to ever need that.  And it could present some coupling issues, so I’d avoid it where possible/feasible and leave the matchers as simple token-like data extraction components.

—jason




On March 11, 2015 at 1:36:40 AM, Tamas Cservenak (notifications@github.com) wrote:

In plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenArtifactMatcher.java:

> +import org.sonatype.sisu.goodies.common.ComponentSupport;
> +
> +/**
> + * Matcher that matches for Maven artifacts only, and sets {@link ArtifactCoordinates} in context attributes.
> + *
> + * @since 3.0
> + */
> +public class MavenArtifactMatcher
> +    extends ComponentSupport
> +    implements Matcher
> +{
> +  @Override
> +  public boolean matches(final Context context) {
> +    final String path = context.getRequest().getPath();
> +    final MavenFacet mavenFacet = context.getRepository().facet(MavenFacet.class);
> +    final ArtifactCoordinates coordinates = mavenFacet.getArtifactCoordinatesParser().parsePath(path);
Am tinkering between the two: have ArtifactCoordinatesParser exposed via facet or not. Originally it was not, and matcher had it injected via ctor, then I exposed it via facet. Will undo, as currently only matcher needs parser, nothing else, no need to have it on a facet (although am unsure about the future).

—
Reply to this email directly or view it on GitHub.